atlasdb-client/src/main/java/com/palantir/atlasdb/stream/StreamStorePersistenceConfiguration.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.stream;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.base.Preconditions;
+
+@JsonSerialize(as = ImmutableStreamStorePersistenceConfiguration.class)
+@JsonDeserialize(as = ImmutableStreamStorePersistenceConfiguration.class)
+@Value.Immutable
+public interface StreamStorePersistenceConfiguration {
+    StreamStorePersistenceConfiguration DEFAULT_CONFIG = ImmutableStreamStorePersistenceConfiguration.builder()
+            .build();
+
+    /**
+     * The number of blocks that a nontransactional storeStream() will store before pausing for
+     * writePauseDurationMillis. For example, if a value of 5 is specified, then storeStream() will
```

Comment:
I don't think "The number of blocks that a nontransactional storeStream() will store before pausing for writePauseDurationMillis" is accurate. This seems to affect both the transactional case and the non-transactional case. Check "storeBlockWithNonNullTransaction" for details.

Reply:
Ah, yes. Confirming this affects both cases.

Reply:
I'll split the code paths - it doesn't make sense for a transactional case to bother with backing off since writes are buffered in memory.